### PR TITLE
ch-signup validation

### PR DESCRIPTION
**Pull request by nwashangai**
#### What does this PR do?
it validates signup input
#### Description of Task to be completed?
* update validation
* write tests
#### How should this be manually tested?
* Not applicable
#### Any background context you want to provide?
* None
#### What are the relevant pivotal tracker stories?
#160702881
#### Screenshots (if appropriate)
* None
#### Questions:
* None

### CODE REVIEW

**Comment by dogoyaro** [comment]
Please title this PR according to the standards we agreed upon.

File: .travis.yml [hunk 1]
```diff
@@ -13,11 +13,15 @@ addons:
...
 before_script:
+  - psql -c 'create database food;' -U postgres
+  - psql -c "CREATE USER young WITH PASSWORD 'andela';" -U postgres
   - travis_wait npm install
   - npm install codeclimate-test-reporter -g
 script:
   - ./node_modules/mocha/bin/mocha --recursive './test/' --compilers js:babel-core/register --exit
+  - psql -c 'drop database food;' -U postgres
 after_success:
+  - psql -c 'create database food;' -U postgres
```
Comment (dogoyaro) [inline .travis.yml:24]: Are you trying to drop this database? Not sure why you're creating after success. Please explain what this does.

File: src/middlewares/validator.js [hunk 2]
```diff
@@ -1,8 +1,11 @@
 import { food } from '../models/store';
+import userModel from '../models/UserModel';
 
 export default (request, response, next) => {
     let isValid = 1;
-    if (request.body.foodItems) {
+    if (request.method === 'POST' && (request.originalUrl === '/api/v1/orders/' || request.originalUrl === '/api/v1/orders')) {
+        const count = (request.body.foodItems) ? (Array.isArray(request.body.foodItems)) ? request.body.foodItems.length : 0 : 0;
```
Comment (dogoyaro) [inline src/middlewares/validator.js]: Aren't if statements more readable than nested ternary operators?

File: src/middlewares/validator.js [hunk 3]
```diff
@@ -1,8 +1,11 @@
...
+        if (!request.body.foodItems === undefined || !count < 1) {
```
Comment (dogoyaro) [inline src/middlewares/validator.js]: I think you only need to check the count here, as the check above should have handled that check. Sync with me if you have questions about this.

File: src/middlewares/validator.js [hunk 4]
```diff
@@ -12,11 +15,31 @@ export default (request, response, next) => {
...
+    } else {
+        isValid = 'No food items';
+    }
         if(isValid === 1) {
             next();
         } else {
             return (response.status(400).json({ status: 'error', message: isValid }));
         }
+    } else if (request.method === 'POST' && (request.originalUrl === '/api/v1/auth/signup' || request.originalUrl === '/api/v1/auth/signup/')) {
+      const regex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;
```
Comment (dogoyaro) [inline src/middlewares/validator.js]: Do you think it might be advantageous to move the logic that validates the email to some kind of helper method, so you can re-use for all the places where you have to validate emails? Perhaps you could even do that for other fields.

File: src/middlewares/validator.js [hunk 5]
```diff
@@ -12,11 +15,31 @@ export default (request, response, next) => {
...
+      isValid = (!regex.test(String(request.body.email)) || !request.body.email) ?
+          'Invalid email id': (!/^[A-Za-z]+$/.test(String(request.body.name)) || !request.body.name) ?
+          'Invalid name': (!request.body.phone || !/^\d{11}$/.test(request.body.phone)) ?
```
Comment (dogoyaro) [inline src/middlewares/validator.js]: Same as above.

File: test/index.js [hunk 6]
```diff
@@ -20,4 +21,13 @@ describe('Fast-Food-Fast', () => {
         });
     });
   });
+  describe('Migration test', () => {
+    it('it should Successfully migrate database schemas', (done) => {
```
Comment (dogoyaro) [inline test/index.js]: This test has no assertions. Please fix.

**Review by dogoyaro — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (nwashangai) [reply .travis.yml:24]: in Travis after running the test, for me to run the cover which has to run another test, I have to recreate the schemas in order not to  duplicate the entry in the tables thereby raising error
